### DISCUSSION TITLE
Would not be better if ble_monitor was just BaseHaRemoteScanner for official HA Bluetooth integration?

### DISCUSSION THREAD

**Discussion by Sorien** [custom-components/ble_monitor#1351] [Ideas]
I have been working on developing my own ESP BLE proxy and have noticed that Home Assistant has a complete infrastructure to support custom "BLE packets providers". In my opinion, it would be more efficient if this project just implemented BaseHaRemoteScanner for HA, and device support was located in the https://github.com/Bluetooth-Devices repository. This would prevent the duplication of effort across two projects or am I missing something obvious?

**Comment by Ernst79** [comment]
No, you’re not missing something obvious, but it is just a matter of history. BLE monitor was started a very long ago, way before any decent Bluetooth support was available in HA. Back then, we decided to use aioblescan to collect BLE data, and developed parsers for many brands in one integration. A few years ago, @bdraco developed proper Bluetooth support in HA, and the HA devs asked me to help moving over support for BLE devices to official HA integrations. Nowadays, we have moved the most important ones, like, xiaomi, govee, qingping, MiScale, BTHome, kegtron and some more. Each with their own integration and parser on the https://github.com/Bluetooth-Devices repository. 

The best would be that we move over all remaining brands as well, but that is a lot of work, that I have to do besides my actual work and maintenance of Xiaomi-ble, BTHome, etc. And that is where the problem is, time, as I am basically the only one maintaining this repo.

So, yes, I totally agree with your suggestion, but i need help to get that done.

**Reply by Sorien** [reply]
thx for your respone, I was just a bit confused, I see that you want to migrate people to official integration but at the same time people still requesting new device integration here, still actively using this addon and after almost two years you are still "actively" developing this addon ... maybe I could find some time and help you move some integrations ...

**Comment by Ernst79** [comment]
The thing is that it is way more easy to add a new device in BLE monitor than setting up a new official integration. But you are right, we should perhaps remove support for devices that are available as official integration nowadays. But I have the feeling that that is not going to be appreciated by everyone :-)
